cudf/bind/cudf_cpp.pxd

```diff
@@ -0,0 +1,1004 @@
+# Copyright (c) 2018, NVIDIA CORPORATION.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+from libcpp cimport bool
+from numpy cimport uint8_t, int64_t, int32_t, int16_t, int8_t
+
+# Utility functions to build gdf_columns, gdf_context and error handling
+
+cpdef _get_ctype_ptr(obj)
+cpdef _get_column_data_ptr(obj)
+cpdef _get_column_valid_ptr(obj)
+
+cdef gdf_column* column_view_from_column(col)
+cdef gdf_column* column_view_from_NDArrays(size, data, mask,
+                                           dtype, null_count)
+
+cdef gdf_context* create_context_view(flag_sorted, method, flag_distinct,
+                                      flag_sort_result, flag_sort_inplace)
+
+cpdef check_gdf_error(errcode)
+
+# Import cudf.h header to import all functions
+# First version of bindings has no changes to the cudf.h header, so this file
+# mirrors the structure in cpp/include
+
+cdef extern from "gdf.h" nogil:
```

Comment:
Is it possible to generate this from the libgdf headers like we did with cffi?

Reply:
yes, it is, a generating script is in the works for that. There are a few kinks we're still working on before adding that script.